Sources/DeepLinking/Route+Component.swift

```diff
@@ -51,82 +51,115 @@ public extension Route {
             }
         }
 
-        // MARK: Key (
+        // MARK: Key
 
-        public enum Key : Hashable{
+        public enum Key {
             case empty
             case constant(String)
             case variable
-
-            public var hashValue: Int {
-                switch self {
-                case .empty: return "\(type(of: self).empty)".hashValue
-                case let .constant(value): return value.hashValue
-                case .variable: return "\(type(of: self).variable)".hashValue
-                }
-            }
-
-            public static func ==(lhs: Key, rhs: Key) -> Bool {
-                switch (lhs, rhs) {
-                case (.empty, .empty), (.variable, .variable): return true
-                case let (.constant(lhsString), .constant(rhsString)): return lhsString == rhsString
-                default: return false
-                }
-            }
         }
+    }
+}
 
-        // MARK: ExpressibleByStringLiteral
+extension Route.Component: Hashable {
 
-        public init(stringLiteral value: String) {
-            self.init(component: value)
-        }
+    // MARK: Hashable
 
-        public init(extendedGraphemeClusterLiteral value: String) {
-            self.init(component: value)
+    public var hashValue: Int {
+        switch self {
+        case .empty: return "\(type(of: self).empty)".hashValue
+        case let .constant(value): return "\(type(of: self).constant)(\(value))".hashValue
+        case let .variable(parameter): return "\(type(of: self).variable)(\(String(describing: parameter)))".hashValue
         }
+    }
 
-        public init(unicodeScalarLiteral value: String) {
-            self.init(component: value)
+    public static func ==(lhs: Route.Component, rhs: Route.Component) -> Bool {
```

Comment:
I tried to check the style guide but it's not clear. I believe operators should be surrounded by a single whitespace when defining them. (For instance, SwiftLint enforces such rule by default). It makes it clearer and easier to perceive the operator format.

Examples (taken from SwiftLint):

```
func <| (lhs: Int, rhs: Int) -> Int {}
func <|< <A>(lhs: A, rhs: A) -> A {}
```

instead of

```
func <|(lhs: Int, rhs: Int) -> Int {}
func <|<<A>(lhs: A, rhs: A) -> A {}
```

Reply:
I think we don't have that rule specified on the style guide, indeed.

However, I think that on declaration the operator can be seen more like a function (which it is), and not as an operator (in the sense that it's being applied to operands that is 😛). 

So by looking at it as a function, in a way it makes sense for me to not have a whitespace before the `(`.

Still, perhaps we can start that discussion on our style guide repo, as like you said SwiftLint doesn't differentiate between the two cases, and when we integrate it we will have to add exceptions on all operator declaration points, if we do it like this (no whitespace before the `(`. We currently have to do that on one of our projects, and it sucks 😓.

Reply:
Ok, so after thinking about this for a while, I can see how it might be better if we have a whitespace before the `(`. This is especially evident when we gave generic operators, where the `<` can create some visual confusion with the operator itself.

Will update the operators 😉